src/dbnode/persist/fs/types.go

```diff
@@ -122,6 +122,11 @@ type DataFileSetReaderStatus struct {
 type DataReaderOpenOptions struct {
 	Identifier  FileSetFileIdentifier
 	FileSetType persist.FileSetType
+	// NB(bodu): This option can inform the reader to not sort
+	// if the downstream consumer will only be reading metadata.
+	// Setting this option will throw an error if a regular `Read()`
+	// is attempted.
+	DoNotSort bool
```

Comment:
nit: Maybe we call this "OptimizedReadMetadataOnly bool" - that way if can do further efficiency gains we can lump them behind the same flag?